dlpar/dlpar_main.py

```diff
@@ -55,105 +62,78 @@ def run_cmd(self, test_cmd, dlpar_type_flag=""):
-            dlpar_type_flag = "cpu_fold"
-            self.log.info(
-                "CPU folding Workload: Calling ./dlpar_workload_setup.py")
-            test_cmd = './dlpar_workload_setup.py'
-            self.run_cmd(test_cmd, "cpu_fold")
-        self.dlpar_engine()
-        if dlpar_type_flag != "":
-            test_cmd = './dlpar_workload_setup.py'
-            if dlpar_type_flag == "smt":
-                self.run_cmd(test_cmd, "smt:kill_process")
-            elif dlpar_type_flag == "cpu_fold":
-                self.run_cmd(test_cmd, "cpu_fold:kill_process")
+        for line in process.system_output('lparstat -i', ignore_status=True,
+                                          shell=True,
+                                          sudo=True).decode("utf-8") \
+                                                    .splitlines():
+            if component in line:
+                a = line.split(':')[-1].strip()
+                print(a)
+                return a
+        return ''
+
+    def setUp(self):
+        self.list_data = []
+        self.lpar_mode = self.params.get('lp_mode', default='dedicated')
+        for i in list_payload:
+            self.data = self.params.get(i, default='')
+            self.list_data.append(self.data)
+
+        # Get HMC IP
+        self.hmc_ip = wait.wait_for(
+            lambda: self.get_mcp_component("HMCIPAddr"), timeout=30)
+
+        # Primary lpar details
+        self.pri_partition = self.get_partition_name("Partition Name")
+        self.pri_name = self.get_partition_name("Node Name")
+        pri_data = {"src_partition": self.pri_partition,
+                    "src_name": self.pri_name,
+                    "hmc_name": self.hmc_ip}
+        self.res = {list_payload[i]: self.list_data[i]
+                    for i in range(len(list_payload))}
+        self.res = dict(list(pri_data.items()) + list(self.res.items()))
+        self.log.info("Calling Config file creation method--!!")
+        self.sorted_payload = dict(sorted(self.res.items()))
+        self.iterations = self.sorted_payload.get('iterations')
+
+    def test_cpu_dlpar(self):
+
+        if self.lpar_mode == 'dedicated':
+            Ded_obj = DedicatedCpu(self.sorted_payload,
+                                   log='dedicated_cpu.log')
+            for i in range(self.iterations):
+                Ded_obj.add_ded_cpu()
+                Ded_obj.rem_ded_cpu()
+                Ded_obj.move_ded_cpu()
+        elif self.lpar_mode == 'shared':
+            Sha_obj = CpuUnit(self.sorted_payload, log='cpu_unit.log')
+            for i in range(self.iterations):
+                Sha_obj.add_proc()
+                Sha_obj.remove_proc()
+                Sha_obj.move_proc()
+
+    def test_mem_dlpar(self):
+        Mem_obj = Memory(self.sorted_payload, log='memory.log')
+        for i in range(self.iterations):
+            Mem_obj.mem_add()
+            Mem_obj.mem_rem()
+            Mem_obj.mem_move()
+            # Mem_obj.mem_mix_ope()
```

Comment:
so the same test will also cover shared scenarios ?

Reply:
mem does not have shared concept